Sources/MediaNavigationBar.swift

```diff
@@ -0,0 +1,135 @@
+        chromeButton.setImage(UIImage(named: "renderer"), for: .normal)
+        chromeButton.tintColor = .white
+        chromeButton.translatesAutoresizingMaskIntoConstraints = false
+        return chromeButton
+    }()
+    
+    lazy var airPlayButton :UIButton = {
+        var airPlayButton = UIButton(type: .system)
+        airPlayButton.addTarget(self, action: #selector(toggleAirPlay), for: .touchUpInside)
+        airPlayButton.setImage(UIImage(named: "iconAirplay"), for: .normal)
+        airPlayButton.tintColor = .white
+        airPlayButton.translatesAutoresizingMaskIntoConstraints = false
+        return airPlayButton
+    }()
+    
+    private func setupConstraints() {
+        NSLayoutConstraint.activate([
+            minimizePlaybackButton.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+            minimizePlaybackButton.topAnchor.constraint(equalTo: self.topAnchor),
+            minimizePlaybackButton.widthAnchor.constraint(equalToConstant: minimizePlaybackButton.frame.width),
+            minimizePlaybackButton.bottomAnchor.constraint(equalTo: self.bottomAnchor),
+            mediaTitleTextLabel.leadingAnchor.constraint(equalTo: minimizePlaybackButton.trailingAnchor, constant: 16),
+            mediaTitleTextLabel.topAnchor.constraint(equalTo: self.topAnchor),
+            mediaTitleTextLabel.trailingAnchor.constraint(equalTo: chromeCastButton.leadingAnchor, constant: 50),
+            chromeCastButton.trailingAnchor.constraint(equalTo: airPlayButton.leadingAnchor, constant: 20),
+            chromeCastButton.topAnchor.constraint(equalTo: self.topAnchor),
+            chromeCastButton.widthAnchor.constraint(equalToConstant: 24),
+            airPlayButton.trailingAnchor.constraint(equalTo: self.trailingAnchor, constant: 20),
+            airPlayButton.topAnchor.constraint(equalTo: self.topAnchor),
+            airPlayButton.widthAnchor.constraint(equalToConstant: 24)
+        ])
+    }
+    
+    private func setupViews() {
+        addSubview(minimizePlaybackButton)
+        addSubview(mediaTitleTextLabel)
+        addSubview(chromeCastButton)
+        addSubview(airPlayButton)
+        
+        if UIApplication.shared.statusBarOrientation.isLandscape {
+            mediaTitleTextLabel.isHidden = true
+        }
+    }
+    
+    // MARK: Initializers
+    required init(coder :NSCoder) {
+        fatalError("init(coder: NSCoder) not implemented")
+    }
+    
+    init() {
+        super.init(frame: .zero)
+        setupViews()
+        setupConstraints()
+    }
+    
+    // MARK: Instance Methods
+    func setMediaTitleLabelText(_ titleText: String) {
+        mediaTitleTextLabel.text = titleText
+    }
+    
+    // MARK: Button Actions
+    func minimizePlayback() {
+        guard let delegate = delegate else {
+            print("Delegate not set for MediaNavigationBar")
+            return
+        }
+    
+        delegate.didMinimizePlayback(self)
```

Comment:
it's simpler to write delegate?.didMinimizePlayback(self)

Reply:
I wanted to at least make debugging down the line a lot easier, so that it would at least print the message to notify that the delegate was not set.

Reply:
if it's just for debugging then we could also just add an assert(delegate != nil) :)
less code = more maintainable. Just as a suggestion but I can also live with the code above